### PR TITLE
feat: implement track detail page

### LINKED ISSUE #512: [Feature]: Track detail page
### Description

`/tracks/[id]`

needs to be implemented

### Explanation

This page is stubbed out already and you are able to enroll.
We want to take the list UI from `/tracks` and use it here for the list items.

see https://leetcode.com/studyplan/top-interview-150 for inspiration

### PR DESCRIPTION
**Pull request by DerTimonius**
## Description

This PR will implement a design for the `/track/slug` pages. This is currently just a first draft to get some feedback on ideas that could be added:

Current looks:
image

## Ideas

- maybe show the shortDescription on hover
- show trackProgress if enrolled

closes #512

### CODE REVIEW

**Comment by bautistaaa** [comment]
- we can show some indicator of `in progress` for challenges they tried.
- needs a progress meter
- a short description on hover is a nice touch
- track name should be aligned with the list items (i dont think the list items need to be so wide
- enroll bottom should be left aligned under the title

**Review by is-it-ayush — CHANGES_REQUESTED** [review]
Thank you for the contribution @DerTimonius. Just a tiny change with the `orderId`, otherwise it's all good! 💙✨

File: apps/web/src/app/[locale]/tracks/_components/track-details.tsx [hunk 1]
```diff
@@ -1,41 +1,85 @@
...
           <EnrollButton action={unenrollUserFromTrack} trackId={trackId} text="Unenroll" />
         )}
       </div>
+      <div className="flex flex-col min-w-[50%] max-w-xl">
+        {challenges
+            .sort((a, b) =>
+              difficultyToNumber[a.difficulty] !== difficultyToNumber[b.difficulty]
+                ? difficultyToNumber[a.difficulty] - difficultyToNumber[b.difficulty]
+                : a.name.localeCompare(b.name),
+            )
```
[the remark this answers is not in the document]
Comment (DerTimonius) [reply apps/web/src/app/[locale]/tracks/_components/track-details.tsx]: Thanks for the feedback, implemented the change [here](https://github.com/typehero/typehero/commit/f65914160a08c6162e3f6d42d331e7971c19f210)

**Comment by bautistaaa** [comment]
we left some comments on the preview url. can you see if you can access it? or even request access to it?

**Comment by DerTimonius** [comment]
@bautistaaa Sorry, can't do either. I only see `Team Access Required You do not have access to comment on this Vercel team’s deployments.` when I try to login

**Comment by bautistaaa** [comment]
> @bautistaaa Sorry, can't do either. I only see `Team Access Required You do not have access to comment on this Vercel team’s deployments.` when I try to login

let me see if i can get you temp access

**Comment by bautistaaa** [comment]
can you give me your email. ill add you to the team temporarily so you can see that feedback comments.

**Comment by DerTimonius** [comment]
timon.jurschitsch@gmail.com

**Comment by bautistaaa** [comment]
> timon.jurschitsch@gmail.com

invite sent

**Comment by DerTimonius** [comment]
sorry for the stupid question, but how do I access the comments? can't find them in the notifications in the previews

**Comment by is-it-ayush** [comment]
> sorry for the stupid question, but how do I access the comments? can't find them in the notifications in the previews

Select "8 unresolved"  from https://github.com/typehero/typehero/pull/594#issuecomment-1704355021 in the "typehero-web-staging" row.
image

**Comment by bautistaaa** [comment]
> > sorry for the stupid question, but how do I access the comments? can't find them in the notifications in the previews
> 
> Select "8 unresolved" from [#594 (comment)](https://github.com/typehero/typehero/pull/594#issuecomment-1704355021) in the "typehero-web-staging" row. image

click visit preview next to that and login

you should see a toolbar bottom middle like his 

CleanShot 2023-09-09 at 08 54 45@2x

**Comment by bautistaaa** [comment]
pulse check on this to see if this PR is still alive

**Comment by DerTimonius** [comment]
yes, still alive. struggling to find a good background (not my expertise tbh)

**Comment by bautistaaa** [comment]
> yes, still alive. struggling to find a good background (not my expertise tbh) help with that is much appreciated!

its okay this doesnt need to be 100% done. incremental progress is fine since we arent live yet!

**Comment by DerTimonius** [comment]
I think other than that I implemented the changes of the comments

**Comment by bautistaaa** [comment]
> I think other than that I implemented the changes of the comments

awesome let me take a look. theres also a conflict that needs resolving

**Comment by is-it-ayush** [comment]
image


Quick Thoughts:
-  The page needs to be slightly more detailed. We need to include description for each challenge such that a user can have a quick glance before enrolling/unenrolling on a track.
- Have an alert before enrolling from the track.
- Search tracks?
- Sort tracks

**Comment by is-it-ayush** [comment]
great work! lgtm 🚀

**Review by is-it-ayush — APPROVED** [review]

**Comment by is-it-ayush** [comment]
Thank you for the PR @DerTimonius! : D

**Comment by DerTimonius** [comment]
Thanks for the review and merge!

**Pull request merged**